pootle/apps/pootle_translationproject/utils.py

```diff
@@ -0,0 +1,148 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) Pootle contributors.
+#
+# This file is a part of the Pootle project. It is distributed under the GPL3
+# or later license. See the LICENSE file for a copy of the license and the
+# AUTHORS file for copyright and authorship information.
+
+from django.contrib.auth import get_user_model
+
+from pootle.core.models import Revision
+from pootle_app.models import Directory
+from pootle_statistics.models import SubmissionTypes
+from pootle_store.diff import StoreDiff
+from pootle_store.models import SOURCE_WINS, Store
+from pootle_translationproject.models import TranslationProject
+
+
+User = get_user_model()
+
+
+class TPTool(object):
+
+    def __init__(self, project):
+        self.project = project
+
+    def get_path(self, language_code):
+        return "/%s/%s/" % (language_code, self.project.code)
+
+    def get_tp(self, language):
+        try:
+            return self.project.translationproject_set.get(
+                language=language)
+        except TranslationProject.DoesNotExist:
+            pass
+
+    def check_tp(self, tp):
+        if tp.project != self.project:
+            raise ValueError(
+                "TP '%s' is not part of project '%s'"
+                % (tp, self.project.code))
+
+    def create_tp(self, language):
+        return self.project.translationproject_set.create(
+            language=language)
+
+    def update_from_tp(self, source, target):
+        self.check_tp(source)
+        self.check_tp(target)
+        self.update_children(source.directory, target.directory)
+
+    def update_store(self, source, target):
+        differ = StoreDiff(target, source, target.get_max_unit_revision() + 1)
+        diff = differ.diff()
+        if diff is None:
+            return
+        system = User.objects.get_system_user()
+        update_revision = Revision.incr()
+        target.update_from_diff(
+            source,
+            target.get_max_unit_revision() + 1,
```

Comment:
can we reuse `target.get_max_unit_revision()` from the first line in this method?